contracts/factories/PairFactory.sol

```diff
@@ -1,40 +1,96 @@
+    bool internal initial_tank_set;
+    address public tank;
+    address public deployer;
+
+    mapping(address => mapping(address => mapping(bool => address)))
+        public getPair;
     address[] public allPairs;
     mapping(address => bool) public isPair; // simplified check if its a pair, given that `stable` flag might not be available in peripherals
 
     address internal _temp0;
     address internal _temp1;
     bool internal _temp;
 
-    event PairCreated(address indexed token0, address indexed token1, bool stable, address pair, uint);
+    event PairCreated(
+        address indexed token0,
+        address indexed token1,
+        bool stable,
+        address pair,
+        uint256
+    );
 
     constructor() {
         pauser = msg.sender;
         isPaused = false;
         feeManager = msg.sender;
-        stableFee = 2; // 0.02%
-        volatileFee = 2;
+        // stableFee = 2; // 0.02%
+        // volatileFee = 2;
+        stableFee = 3; // 0.03%
+        volatileFee = 25; // 0.25%
+        deployer = msg.sender;
+    }
+
+    // need to set team so that team can set voter we really only need to set the voter once :) - deployer can do this only once (either in the init script but then we have to do this one last otherwise it will fail)
+
+    function setTeam(address _team) external {
+        require(msg.sender == deployer); // might need to set this to deployer?? or just make it
+        require(team == address(0), "The team has already been set.");
+        team = _team;
+    }
+
+    // In this example, the owner variable is set to the address that deploys the contract in the constructor. The setTeam function requires that the caller (msg.sender) must be the owner, and that the team variable has not yet been set (it is equal to the address value of 0). If either of these conditions are not met, the function will revert and not update the team variable.
+
+    // we only get once shot at this.
+
+    function setVoter(address _voter) external {
+        require(!initial_voter_set, "The voter has already been set.");
+        // require(msg.sender == deployer); // have to make sure that this can be set to the voter addres during init script
+        voter = _voter;
+        initial_voter_set = true;
+    }
+
+    // function set tank on factory require team
+
+    function setTank(address _tank) external {
+        require(!initial_tank_set, "The tank has already been set.");
+        // require(msg.sender == deployer); // this should be updateable to team but adding deployer so that init script can run..
+        tank = _tank;
+        initial_tank_set = true;
+    }
+
+    // This makes tank updateable forever by the team address (multisig)
+
+    function acceptTank(address _tank) external {
+        require(msg.sender == team, "not pending team");
+        tank = _tank;
+    }
+
+    // pair uses this to check if voter is updating external_bribe
+
+    function getVoter() external view returns (address) {
```

Comment:
we don't need this function.

Just call `PairFactory(address).voter()` in `Pair`